mne/io/otb/otb.py

```diff
@@ -0,0 +1,251 @@
+
+import tarfile
+from collections import Counter
+from datetime import datetime
+from pathlib import Path
+
+import numpy as np
+from defusedxml import ElementTree as ET
+
+from ..._fiff.meas_info import create_info
+from ...utils import _check_fname, fill_doc, logger, verbose, warn
+from ..base import BaseRaw
+
+OTB_PLUS_DTYPE = np.int16  # the file I have is int16; TBD if *all* OTB+ are like that
+FORMAT_MAPPING = dict(
+    d="double",
+    f="single",
+    i="int",
+    h="short",
+)
+OTB_PLUS_FORMAT = FORMAT_MAPPING[OTB_PLUS_DTYPE().dtype.char]
+
+
+def _parse_patient_xml(tree):
+    """Convert an ElementTree to a dict."""
+
+    def _parse_sex(sex):
+        # TODO English-centric; any value of "sex" not starting with "m" or "f" will get
+        # classed as "unknown". TBD if non-English-like values in the XML are possible
+        # (e.g. maybe the recording GUI is localized but the XML values are not?)
+        return dict(m=1, f=2)[sex.lower()[0]] if sex else 0
+
+    def _parse_date(dt):
+        return datetime.fromisoformat(dt).date()
+
+    subj_info_mapping = (
+        ("family_name", "last_name", str),
+        ("first_name", "first_name", str),
+        ("weight", "weight", float),
+        ("height", "height", float),
+        ("sex", "sex", _parse_sex),
+        ("birth_date", "birthday", _parse_date),
+    )
+    subject_info = dict()
+    for source, target, func in subj_info_mapping:
+        if value := tree.find(source):
+            subject_info[target] = func(value.text)
+    return subject_info
+
+
+@fill_doc
+class RawOTB(BaseRaw):
+    """Raw object from an OTB file.
+
+    Parameters
+    ----------
+    fname : path-like
+        Path to the OTB file.
+    %(verbose)s
+
+    See Also
+    --------
+    mne.io.Raw : Documentation of attributes and methods.
+    """
+
+    @verbose
+    def __init__(self, fname, *, verbose=None):
+        # Adapted from the MATLAB code at:
+        # https://github.com/OTBioelettronica/OTB-Matlab/tree/main/MATLAB%20Open%20and%20Processing%20OTBFiles
+        # with permission to relicense as BSD-3 granted here:
+        # https://github.com/OTBioelettronica/OTB-Python/issues/2#issuecomment-2979135882
+        fname = str(_check_fname(fname, "read", True, "fname"))
+        otb_version = "four" if fname.endswith(".otb4") else "plus"
+        logger.info(f"Loading {fname}")
+
+        self.preload = True  # lazy loading not supported
+
+        highpass = list()
+        lowpass = list()
+        ch_names = list()
+        ch_types = list()
+
+        # TODO verify these are the only non-data channel IDs
+        NON_DATA_CHS = ("Quaternion", "BufferChannel", "RampChannel")
+
+        with tarfile.open(fname, "r") as fid:
+            fnames = fid.getnames()
+            # the .sig file is the binary channel data
+            sig_fname = [_fname for _fname in fnames if _fname.endswith(".sig")]
+            assert len(sig_fname) == 1  # TODO is this a valid assumption?
```

Comment:
is there guaranteed to be only 1 `.sig` file in each OTB+ archive?  The MATLAB code suggests "yes" but it's not totally clear.